pipelines/youtube_data/__init__.py

```diff
@@ -0,0 +1,181 @@
+import datetime
+from typing import Any, Dict, Iterator, List, Sequence
+
+from googleapiclient.discovery import build
+
+import dlt
+from dlt.extract.source import DltResource, DltSource
+from dlt.common.typing import TDataItem
+
+
+_SERVICE_NAME = "youtube"
+_SERVICE_VERSION = "v3"
+current_time = datetime.datetime.now()
+
+def _get_channel_id(youtube: Any, channel_name: str) -> List[str]:
+    """
+    function to retrieve channel id given channel custom name e.g. MrBeast
+
+    params:
+        youtube: YouTube resource object
+        channel_name: YouTube custom channel name
+
+    return:
+        channel_id: YouTube channel id
+    """
+    response = youtube.search().list(
+        part="snippet",
+        q={channel_name},
+        type="channel"
+    ).execute()
+
+    for item in response["items"]:
+        #filter only youtube channel
+        if item["id"]["kind"] == "youtube#channel":
+            return item["id"]["channelId"]
+
+def _get_channel_video(
+        youtube: Any, 
+        channel_id: List[str], 
+        max_results: int,
+        start_date: str,
+        end_date: str
+) -> List[str]:
+    """
+    function to retrieve all published videos given channel id
+
+    params:
+        youtube: YouTube resource object
+        channel_id: YouTube channel id
+        max_results: max results to be displayed each API call
+        start_date: start date of published video
+        end_date: last date of published video
+
+    return:
+        video_ids: list of videos ids
+    """
+    video_ids = []
+    next_page_token = None
+
+    while True:
+        response = youtube.search().list(
+            channelId=channel_id,
+            part="id",
+            maxResults=max_results,
+            publishedAfter=start_date,
+            publishedBefore=end_date,
+            page_token=next_page_token
+        ).execute()
+
+        for item in response['items']:
+            if 'videoId' in item['id']:
+                video_id = item['id']['videoId']
+                video_ids.append(video_id)
+        
+        # check if there are next page results
+        if 'nextPageToken' not in response:
+            break
+
+    return video_ids
+
+def _get_video_details(youtube: Any, video_list: List[str], max_results: int) -> Dict[str, str]:
```

Comment:
This function returns a list of dictionaries, but the type hint is `Dict[str, str]`.